src/Microsoft.DotNet.Wpf/src/System.Xaml/System/Xaml/XamlSchemaContext.cs

```
@@ -1176,7 +1176,14 @@ bool UpdateNamespaceByUriList(XmlNsInfo nsInfo)
             IList<XmlNsInfo.XmlNsDefinition> xmlnsDefs = nsInfo.NsDefs;
             foreach (XmlNsInfo.XmlNsDefinition xmlnsDef in xmlnsDefs)
             {
-                AssemblyNamespacePair pair = new AssemblyNamespacePair(nsInfo.Assembly, xmlnsDef.ClrNamespace);
+                Assembly assembly = nsInfo.Assembly;
+                if (!string.IsNullOrEmpty(xmlnsDef.AssemblyName))
+                {
+                    assembly = ResolveAssembly(xmlnsDef.AssemblyName);
+                    if(assembly == null)
+                        return false;
```

Is this the correct action here rather than `continue` or `throw`?